### PR TITLE
Refactor default entropy and fix test

### LINKED ISSUE #96: Use init() instead of sync.Once to initiate default entropy
In the `DefaultEntropy` you are using `sync.Once` to initiate the entropy and new reader. This means that every call to `Make()` has to make atomic call to see if `sync.Once` has been invoked. This seems very pointless when you could have simply used `init()`, which is run only once, and poses no performance hit. Especially when this is not ephemeral code but rather eternal and called many many many times.

### PR DESCRIPTION
**Pull request by peterbourgon**
Fixes #96 and #97.

### CODE REVIEW

File: ulid_test.go [hunk 1]
```diff
@@ -621,7 +621,8 @@ func TestMonotonicSafe(t *testing.T) {
 	t.Parallel()
 
 	var (
-		safe = ulid.DefaultEntropy()
+		rng  = rand.New(rand.NewSource(time.Now().UnixNano()))
+		safe = &ulid.LockedMonotonicReader{MonotonicReader: ulid.Monotonic(rng, 0)}
```
Comment (peterbourgon) [inline ulid_test.go:625]: Re-use of default entropy among parallel tests was creating test flakes when the per-ms entropy ceiling was hit.

Comment (tsenart) [reply ulid_test.go:625]: Help me understand please. What kind of flakes? The default entropy should be safe. If it's not, that's a problem?

**Review by tsenart — COMMENTED** [review]
Sorry, this fell through the cracks

File: ulid_test.go (hunk 1 above)
Comment (peterbourgon) [reply ulid_test.go:625]: It's definitely safe. When the tests were running with a high level of concurrency, it was possible to make so many Reads from the DefaultEntropy in a single millisecond (from different tests) that it no longer provided monotonicity.

Comment (peterbourgon) [reply ulid_test.go:625]: Strictly speaking, the documentation for [Monotonic](https://pkg.go.dev/github.com/oklog/ulid/v2#Monotonic) is incorrect when it says

> Monotonic returns an entropy source **that is guaranteed to yield strictly increasing entropy bytes for the same ULID timestamp**. On conflicts, the previous ULID entropy is incremented with a random number between 1 and `inc` (inclusive).

Rather, the entropy will be strictly increasing **until** adding that random number goes beyond the available entropy space. That's the limit we hit.

Comment (tsenart) [reply ulid_test.go:625]: OK, LGTM

**Pull request merged**